style.md

```diff
@@ -811,34 +813,43 @@ seconds that may have occurred between those two instants.
 
 <!-- TODO: section on String methods for enums -->
 
-### Error Types
+### Errors
 
-There are various options for declaring errors:
+#### Error Types
 
-- [`errors.New`] for errors with simple static strings
-- [`fmt.Errorf`] for formatted error strings
-- Custom types that implement an `Error()` method
-- Wrapped errors using [`"pkg/errors".Wrap`]
+There are few options for declaring errors.
+Consider the following before picking the option best suited for your use case.
 
-When returning errors, consider the following to determine the best choice:
+- Does the caller need to match the error so that they can handle it?
+  If yes, we must support the [`errors.Is`] and [`errors.As`] functions
+  by declaring a top-level error variable or a custom type.
+- Is the error message a static string,
+  or is it a dynamic string that requires contextual information?
+  For the former, we can use [`errors.New`], but for the latter we must
+  use [`fmt.Errorf`] or a custom error type.
+- Are we propagating a new error returned by a downstream function?
+  If so, see the [section on error wrapping](#error-wrapping).
```

Comment:
semi-meta, but the perspective here is first person; ideally it's consistent throughout the guide (it's not atm). we should think about what perspective makes sense/is the most accessible to read.

Reply:
Yeah. It's been a mix of "do this" and "we should do this" and "this should be done".